### PR TITLE
The new India data provider was not updating the state_wise_daily.csv.

### PR DESCRIPTION
**Pull request by themonk911**
Switch to the states.csv file instead, and update code to handle new data format.

Also add some missing union territories and districts to metadata.csv.
Improve matching by correcting some typos in the source data on ingestion.

### CODE REVIEW

File: src/data/metadata.csv [hunk 1]
```diff
@@ -13246,6 +13250,7 @@ IN_TN_MDR,IN,India,TN,Tamil Nadu,MDR,Madurai,,,,
 IN_TN_NIL,IN,India,TN,Tamil Nadu,NIL,Nilgiris,,,The Nilgiris,
 IN_TN_NMK,IN,India,TN,Tamil Nadu,NMK,Namakkal,,,,
 IN_TN_NPM,IN,India,TN,Tamil Nadu,NPM,Nagapattinam,,,,
+IN_TN_MDT,IN,India,TN,Tamil Nadu,MDT,Mayiladuthurai,,,,
```
Comment (owahltinez) [inline src/data/metadata.csv:13253]: This should probably be moved up. From the unit test error logs: `AssertionError: 'IN-TN-MDT' not greater than 'IN-TN-NPM' : Keys in metadata.csv must follow lexicographical order: IN-TN-MDT ≤ IN-TN-NPM`

**Review by owahltinez — CHANGES_REQUESTED** [review]

File: src/data/metadata.csv (hunk 1 above)
Comment (themonk911) [reply src/data/metadata.csv:13253]: Ah thank you!

Comment (themonk911) [reply src/data/metadata.csv:13253]: Done.

**Comment by themonk911** [comment]
As far as I can tell, the test is now failing on the French data source.

**Review by owahltinez — APPROVED** [review]
LGTM. I'm re-running the unit tests one more time. I'll merge by EOD even if we can't get a pass due to the flaky tests.

**Pull request merged**